src/terms/pairwise.jl

```diff
@@ -0,0 +1,131 @@
+    V
+    params
+    max_radius
+end
+function PairwisePotential(V, params; max_radius=100)
+    PairwisePotential(V, params, max_radius)
+end
+(P::PairwisePotential)(basis::AbstractBasis) = TermPairwisePotential(P.V, P.params, P.max_radius, basis)
+
+struct TermPairwisePotential{TV, Tparams, T} <:Term
+    V::TV
+    params::Tparams
+    max_radius::T
+    energy::Real
+end
+function TermPairwisePotential(V, params, max_radius, basis::PlaneWaveBasis{T}) where {T}
+    TermPairwisePotential(V, params, max_radius, energy_pairwise(basis.model, V, params, max_radius))
+end
+
+function ene_ops(term::TermPairwisePotential, basis::PlaneWaveBasis, ψ, occ; kwargs...)
+    (E=term.energy, ops=[NoopOperator(basis, kpt) for kpt in basis.kpoints])
+end
+
+@timing "forces: Pairwise" function compute_forces(term::TermPairwisePotential,
+                                                   basis::PlaneWaveBasis{T}, ψ, occ;
+                                                   kwargs...) where {T}
+    atoms = basis.model.atoms
+    forces_pairwise = zeros(Vec3{T}, sum(length(positions) for (elem, positions) in atoms))
+    energy_pairwise(basis.model, term.V, term.params, term.max_radius; forces=forces_pairwise)
+    # translate to the "folded" representation
+    f = [zeros(Vec3{T}, length(positions)) for (type, positions) in atoms]
+    count = 0
+    for i = 1:length(atoms)
+        for j = 1:length(atoms[i][2])
+            count += 1
+            f[i][j] += forces_pairwise[count]
+        end
+    end
+    @assert count == sum(at -> length(at[2]), atoms)
+    f
+end
+
+function energy_pairwise(model::Model, V, params, max_radius; kwargs...)
+    atom_types = [element.symbol for (element, positions) in model.atoms for _ in positions]
+    positions = [pos for (_, positions) in model.atoms for pos in positions]
+
+    energy_pairwise(model.lattice, atom_types, positions, V, params, max_radius; kwargs...)
+end
+
+"""
+Compute the pairwise interaction energy per unit cell between atomic sites. If `forces` is
+not nothing, minus the derivatives of the energy with respect to `positions` is computed.
+The potential is expected to decrease quickly at infinity.
+"""
+function energy_pairwise(lattice, atom_types, positions, V, params, max_radius; forces=nothing)
+    T = eltype(lattice)
+    @assert length(atom_types) == length(positions)
+
+    if forces !== nothing
+        @assert size(forces) == size(positions)
+        forces_pairwise = copy(forces)
+    end
+
+    # Function to return the indices corresponding
+    # to a particular shell.
+    # Not performance critical, so we do not type the function
+    max_shell(n, trivial) = trivial ? 0 : n
+    # Check if some coordinates are not used.
+    is_dim_trivial = [norm(lattice[:,i]) == 0 for i=1:3]
+    function shell_indices(nsh)
+        ish, jsh, ksh = max_shell.(nsh, is_dim_trivial)
+        [[i,j,k] for i in -ish:ish for j in -jsh:jsh for k in -ksh:ksh
+         if maximum(abs.([i,j,k])) == nsh]
+    end
```

Comment:
This and the loop over shells can probably easily be factored out and used both here and for the ewald.

Reply:
I tried, but ran into problems due to the use of `atom_types`. It would slow down Ewald as a result.

Reply:
as discussed, I'm ok if you add a respective comment in your code and the ewald.